tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -4961,6 +4961,159 @@ TEST_P(OpConverter_FP32_Test, ConvertSqueeze) {
   }
 }
 
+TEST_P(OpConverter_FP32_FP16_INT32_Test, ConvertStridedSlice2) {
+  // Get nodedef for StridedSlice layer.
+  auto get_strided_slice_nodedef =
+      [](DataType tf_type, int64 begin_mask = 0, int64 end_mask = 0,
+         int64 ellipsis_mask = 0, int64 new_axis_mask = 0,
+         int64 shrink_axis_mask = 0) -> NodeDef {
+    Scope s = Scope::NewRootScope();
+    auto input = ops::Placeholder(s.WithOpName("input"), tf_type);
+    auto begin = ops::Placeholder(s.WithOpName("begin"), DT_INT32);
+    auto end = ops::Placeholder(s.WithOpName("end"), DT_INT32);
+    auto strides = ops::Placeholder(s.WithOpName("strides"), DT_INT32);
+    ops::StridedSlice::Attrs attrs = ops::StridedSlice::Attrs()
+                                         .BeginMask(begin_mask)
+                                         .EndMask(end_mask)
+                                         .EllipsisMask(ellipsis_mask)
+                                         .NewAxisMask(new_axis_mask)
+                                         .ShrinkAxisMask(shrink_axis_mask);
+    auto strided_slice = ops::StridedSlice(s.WithOpName("my_strided_slice"),
+                                           input, begin, end, strides, attrs);
+    return strided_slice.operation.node()->def();
+  };
+  struct TestParams {
+    std::vector<int> input_dims;
+    std::vector<int> begin;
+    std::vector<int> end;
+    std::vector<int> strides;
+    int begin_mask;
+    int end_mask;
+    int ellipsis_mask;
+    int new_axis_mask;
+    int shrink_axis_mask;
+    std::vector<int> expected_output_dims;
+    std::vector<float> expected_output;
+    Status conversion_status;
+    Status runtime_status;
+    std::vector<int> partial_input_dims;
+  };
+  auto get_mask = [](const std::vector<int>& mask) {
+   int result = 0;
+   for (int i = 0; i < mask.size(); i++) {
+     if (mask[i]) result += (1 << i);
+   }
+   return result;
+ };
```

Comment:
Please remove code duplication: you should reuse code for ConvertStridedSlice and ConvertStridedSlice2 tests.